runtime/sema/check_interface_declaration.go

```diff
@@ -33,6 +33,12 @@ import (
 // through `declareInterfaceMembers`.
 func (checker *Checker) VisitInterfaceDeclaration(declaration *ast.InterfaceDeclaration) (_ struct{}) {
 
+	wasInInterface := checker.inInterface
+	checker.inInterface = true
+	defer func() {
+		checker.inInterface = wasInInterface
+	}()
```

Comment:
This was removed in https://github.com/dapperlabs/cadence-internal/pull/179, but had to add this back again, since entitlements were relying on it.
We should double-check if an issue similar to the original one also exists with entitlements. cc: @dsainati1